src/types.rs

```diff
@@ -306,75 +348,146 @@ impl Signature {
+    ///     (2, Some("A".to_string())),
+    ///     (1, Some("B".to_string())),
+    ///     (0, Some("C".to_string())),
+    /// ]);
     ///
-    /// let sigchange = sig1.merge(sig2, MergeStrategy::SameOperators);
-    /// // we only reify terms/rules/TRSs associated with sig2
-    /// let trs = sigchange.reify_trs(trs);
-    /// // now term and rule both exist with symbols according to sig1.
-    /// ```
-    /// [`Signature`]: struct.Signature.html
-    /// [`SignatureChange`]: struct.SignatureChange.html
-    pub fn merge(&mut self, mut other: Signature, strategy: MergeStrategy) -> SignatureChange {
-        let delta_op = match strategy {
-            MergeStrategy::SameOperators => 0,
-            MergeStrategy::OperatorsByArityAndName => {
-                let old_len = self.operators.len();
-                for op_spec in other.operators {
-                    if !self.operators.contains(&op_spec) {
-                        self.operators.push(op_spec)
+    /// sig1.merge(sig2, MergeStrategy::DistinctOperators);
+    ///
+    /// let ops: Vec<String> = sig1.operators().iter().map(|op| op.display(&sig1)).collect();
+    ///
+    /// assert_eq!(ops, vec![".", "S", "K", "A", "B", "C"]);
+    ///
+    /// let (mut sig1, _) = Signature::new(vec![
+    ///     (2, Some(".".to_string())),
+    ///     (0, Some("S".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// let (mut sig2, _) = Signature::new(vec![
+    ///     (2, Some(".".to_string())),
+    ///     (0, Some("S".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// sig1.merge(sig2, MergeStrategy::SameOperators);
+    ///
+    /// let ops: Vec<String> = sig1.operators().iter().map(|op| op.display(&sig1)).collect();
+    ///
+    /// assert_eq!(ops, vec![".", "S", "K"]);
+    ///
+    /// let (mut sig1, _) = Signature::new(vec![
+    ///     (2, Some(".".to_string())),
+    ///     (0, Some("S".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// let (mut sig2, _) = Signature::new(vec![
+    ///     (2, Some(".".to_string())),
+    ///     (1, Some("S".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// assert!(sig1.merge(sig2, MergeStrategy::SameOperators).is_err());
+    ///
+    ///    
+    /// let (mut sig1, _) = Signature::new(vec![
+    ///     (2, Some(".".to_string())),
+    ///     (0, Some("S".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// let (mut sig2, _) = Signature::new(vec![
+    ///     (2, Some("A".to_string())),
+    ///     (1, Some("B".to_string())),
+    ///     (0, Some("K".to_string())),
+    /// ]);
+    ///
+    /// sig1.merge(sig2, MergeStrategy::OperatorsByArityAndName);
+    ///
+    /// let ops: Vec<String> = sig1.operators().iter().map(|op| op.display(&sig1)).collect();
+    ///
+    /// assert_eq!(ops, vec![".", "S", "K", "A", "B"]);    
+    /// ```
```

Comment:
Wow - nice test! Could you please just add a comment when you start setting up the test of the next `MergeStrategy`? That might make this easier to parse.